gitpod-io/gitpod-vscode-desktop
# Use new `getSSHPublicKeys` api method

## Description

Improve ssh connection by using the new `getSSHPublicKeys` api method, owner token is no longer needed as part of the remote uri authority.

## How to test

1. [Run this extension](https://github.com/gitpod-io/gitpod-vscode-desktop/blob/master/docs/CONTRIBUTING.md)
2. Create a workspace in this prev env https://jp-vscode-ab4260534a.preview.gitpod-dev.com/workspaces (https://github.com/gitpod-io/gitpod/pull/11246) and connect to vscode desktop
3. If you don't have any SSH keys registered in your gitpod account then it will tell you to use the ownertoken
4. Register an SSH key in your gitpod account, and try connecting again, it should connect directly

Questions:
- Should we keep previous behavior and add a migration path to the new behavior? As it is right now all current users connecting SSH gateway (not that many) will be forced to connect using ownertoken until they register an SSH key in their Gitpod account.

## Release Notes

```release-note
NONE
```
